hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,576 @@
+	//              ToleranceBase::createSymmetric<OrientationTolerance>(rx, ry, 0, 0, 0, 2.0 * M_PI)),
+	//             0.0, discretization));
+ //  	}
+
+	// // Translates a point relative to a reference pose to an absolute transformation
+	// // Also flips the z axis of the orientation to point INTO the plane specified by the
+	// // reference frame.
+	// tf::Transform HybridPlanner::createNominalTransform(const geometry_msgs::Pose& ref_pose, const geometry_msgs::Point& point)
+	// {
+	// 	// To plane
+	// 	tf::Transform marker_pose;
+	// 	tf::poseMsgToTF(ref_pose, marker_pose);
+
+	// 	// From plane to point
+	// 	tf::Transform to_point = tf::Transform::getIdentity();
+	// 	to_point.setOrigin(tf::Vector3(point.x, point.y, point.z));
+	// 	// Reverse orientation of z axis
+	// 	tf::Quaternion quat;
+	// 	quat.setEuler(0.0, M_PI, 0.0); // yaw, pitch roll
+	// 	tf::Transform flip_z (quat);
+	// 	// Calculate transform
+	// 	tf::Transform in_world = marker_pose * to_point * flip_z;
+
+	// 	return in_world;
+	// }
+
+	// void HybridPlanner::moveit_computeCartesianPath(const std::vector<geometry_msgs::Pose>& waypoints)
+	// {
+	// 	double result = group.computeCartesianPath(const std::vector<geometry_msgs::Pose> &waypoints, double eef_step, double jump_threshold,
+ 	//                  moveit_msgs::RobotTrajectory &trajectory,  bool avoid_collisions = true);
+	// }
+
+	//TODO stick to one naming scheme 
+	void HybridPlanner::planProcessPath()
+	{	
+		//TOCHECK is this necessary. This is for a special case of Axial_Symm_Points. 
+		//TOCHECK Make demos for various types (blending, etc?) i.e. (rot symm, etc)
+		std::vector<double> start_pose, end_pose;
+		if(descartes_traj_input_.front()->getClosestJointPose(config_.seed_pose,*descartes_robot_model_ptr_,start_pose) &&
+		descartes_traj_input_.back()->getClosestJointPose(config_.seed_pose,*descartes_robot_model_ptr_,end_pose))
+		{
+		ROS_INFO_STREAM("Setting trajectory start and end to JointTrajectoryPts");
+
+		// Creating Start JointTrajectoryPt from start joint pose
+		descartes_core::TrajectoryPtPtr start_joint_point = descartes_core::TrajectoryPtPtr(
+		new descartes_trajectory::JointTrajectoryPt(start_pose));
+
+		// Creating End JointTrajectoryPt from end joint pose
+		descartes_core::TrajectoryPtPtr end_joint_point = descartes_core::TrajectoryPtPtr(
+		new descartes_trajectory::JointTrajectoryPt(end_pose));
+
+		// Modifying start and end of the trajectory.
+		descartes_traj_input_[0] = start_joint_point;
+		descartes_traj_input_[descartes_traj_input_.size() - 1 ] = end_joint_point;
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to find closest joint pose to seed pose at the start or end of trajectory");
+		exit(-1);
+		}
+
+		//TOCHECK Should the planners be initialized/modified in here instead of init()
+		//as the RobotPtr might have changed in its internal parameters
+
+		if(config_.descartes_planner_type == "dense")
+		{
```

Comment:
Should the planners be initialized/modified in here instead of init() as the RobotPtr might have changed in its internal parameters